test/tap/tests/Makefile

```diff
@@ -117,3 +118,5 @@ galera_2_timeout_no_count: galera_2_timeout_no_count.cpp $(TAP_LIBDIR)/libtap.a
 aurora: aurora.cpp $(TAP_LIBDIR)/libtap.a
 	g++ -DTEST_AURORA -DDEBUG aurora.cpp ../tap/SQLite3_Server.cpp $(INCLUDEDIRS) $(LDIRS) $(OPT) -std=c++11 $(OBJ) $(MYLIBS) -ltap -ldl -o aurora -DGITVERSION=\"$(GIT_VERSION)\"
 
+set_testing-t: set_testing-t.cpp $(TAP_LIBDIR)/libtap.a
+	g++ set_testing-t.cpp -Wall $(INCLUDEDIRS) $(LDIRS) $(OPT) -std=c++11 $(MYLIBS) -ltap -pthread -O0 -ggdb -ldl -lssl -lcrypto -o set_testing-t
```

Comment:
In more recent versions of GCC, for example, the one I'm using (9.2.1), I think it's required to use "-Wl,--no-as-needed" prior to "-ldl", otherwise, linker wont be instructed to link against "dl". (you know --no-as-needed is not required, but it's handy).

Reply:
```
g++ (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
```
$ cat /etc/debian_version
9.11
```
Should I upgrade my compiler?

Reply:
I don't know, I don't know which specific version of GCC started behaving like this, but maybe it's worth trying to check with another version and see if they build, I will try and report you back.